backend/dal/Helpers/Extensions/QueryableExtensions.cs

```diff
@@ -0,0 +1,161 @@
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public static IQueryable<T> OrderByProperty<T>(this IQueryable<T> source, params string[] propertyName)
+        {
+            if (propertyName == null) return source;
+            var query = source;
+            foreach (var prop in propertyName)
+            {
+                var parts = prop?.Split(' ') ?? throw new ArgumentNullException(nameof(propertyName));
+
+                if (parts.Length > 2) throw new ArgumentOutOfRangeException(nameof(propertyName), "Argument 'propertyName' must not have more than two parts (i.e. 'Name asc' or 'Name desc')");
+                if (parts.Length == 2 && _orderByDescending.Contains(parts[1].ToLower()))
+                {
+                    query = query.OrderByPropertyDescending(parts[0]);
+                }
+                else
+                {
+                    var rType = GetPropertyPathType<T>(parts[0]) ?? throw new InvalidOperationException($"The property path '{typeof(T).Name}.{parts[0]}' did not return a type.");
+                    var convertMethod = GeneratePropertyPathLambdaMethod.MakeGenericMethod(new[] { typeof(T), rType });
+                    var orderExpression = convertMethod.Invoke(null, new[] { parts[0] });
+                    if (orderExpression != null)
+                    {
+                        var genericMethod = OrderByMethod.MakeGenericMethod(typeof(T), rType);
+                        query = (IQueryable<T>)genericMethod.Invoke(null, new[] { query, orderExpression });
+                    }
+                    else
+                    {
+                        if (!PropertyExists<T>(parts[0])) continue;
+                        ParameterExpression paramterExpression = Expression.Parameter(typeof(T));
+                        Expression orderByProperty = Expression.Property(paramterExpression, parts[0]);
+                        LambdaExpression lambda = Expression.Lambda(orderByProperty, paramterExpression);
+                        MethodInfo genericMethod = OrderByMethod.MakeGenericMethod(typeof(T), orderByProperty.Type);
+                        query = (IQueryable<T>)genericMethod.Invoke(null, new object[] { query, lambda });
+                    }
+                }
+
+            }
+            return query;
+        }
+
+        /// <summary>
+        /// Order the query results by the specified property names.
+        /// Any property name that doesn't exist will be ignored.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="propertyName"></param>
+        /// <returns></returns>
+        public static IQueryable<T> OrderByPropertyDescending<T>(this IQueryable<T> source, params string[] propertyName)
```

Comment:
Provides a way to dynamically handle sorting by an array of property path values.